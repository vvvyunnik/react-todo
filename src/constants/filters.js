const FilterType = Object.freeze({
  None: "None",
  Completed: "Completed",
  Active: "Active",
});

const FilterLabelMap = Object.freeze({
  [FilterType.None]: "ALL",
  [FilterType.Completed]: "COMPLETED",
  [FilterType.Active]: "ACTIVE",
});

const TodoFilters = [FilterType.None, FilterType.Completed, FilterType.Active];

export { FilterType, FilterLabelMap, TodoFilters };
