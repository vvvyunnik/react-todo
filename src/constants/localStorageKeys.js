const LocalStorageKeys = Object.freeze({
  Todo: "todo-items",
});

export default LocalStorageKeys;
