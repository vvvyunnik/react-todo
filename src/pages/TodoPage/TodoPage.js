import TodoList from "../../components/TodoList/TodoList";
import TodoInput from "../../components/TodoInput/TodoInput";
import Header from "../../components/Header/Header";
import TodoPanel from "../../components/TodoPanel/TodoPanel";
import "./TodoPage.scss";

function TodoPage() {
  return (
    <div className="todo-page">
      <Header />
      <div className="todo-container">
        <TodoInput className="todo-input" />
        <div className="todo-area">
          <TodoList />
          <TodoPanel />
        </div>
      </div>
    </div>
  );
}

export default TodoPage;
