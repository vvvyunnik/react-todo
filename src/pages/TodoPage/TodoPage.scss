.todo-page {
    display: flex;
    flex-flow: column;
    height: 100%;
    justify-content: center;

    .todo-container {
        display: flex;
        flex-flow: column;
        align-items: center;
        height: 100%;
        position: relative;
        top: -20px;

        .todo-input {
            width: 35%;
        }

        .todo-area {
            width: 35%;
            margin-top: 10px;
        }
    }
}