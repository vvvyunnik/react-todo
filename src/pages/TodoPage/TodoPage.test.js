import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { FilterType } from "../../constants/filters";
import TodoPage from "./TodoPage";

const mockStore = configureStore([]);

describe("TodoPage", () => {
  afterEach(cleanup);

  test("matches snapshot", () => {
    const store = mockStore({
      todo: { items: [] },
      filter: { filter: FilterType.None },
    });
    const todoElement = render(
      <Provider store={store}>
        <TodoPage />
      </Provider>
    );

    expect(todoElement).toMatchSnapshot();
  });
});
