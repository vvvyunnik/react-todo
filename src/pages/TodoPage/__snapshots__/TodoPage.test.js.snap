// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoPage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo-page"
      >
        <header
          class="header"
        >
          <h1>
            TODO
          </h1>
        </header>
        <div
          class="todo-container"
        >
          <input
            class="todo-input"
            placeholder="Add a new todo"
            type="text"
            value=""
          />
          <div
            class="todo-area"
          >
            <div
              class="todos"
            >
              <p
                class="no-item-text"
              >
                No todos
              </p>
            </div>
            <div
              class="todo-panel"
            >
              <span
                class="todo-active-count"
              >
                0
                 items left
              </span>
              <div>
                <button
                  class="no-style-btn selected"
                  type="button"
                >
                  ALL
                </button>
                <button
                  class="no-style-btn "
                  type="button"
                >
                  COMPLETED
                </button>
                <button
                  class="no-style-btn "
                  type="button"
                >
                  ACTIVE
                </button>
              </div>
              <button
                class="no-style-btn clear-btn"
                type="button"
              >
                Clear completed
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo-page"
    >
      <header
        class="header"
      >
        <h1>
          TODO
        </h1>
      </header>
      <div
        class="todo-container"
      >
        <input
          class="todo-input"
          placeholder="Add a new todo"
          type="text"
          value=""
        />
        <div
          class="todo-area"
        >
          <div
            class="todos"
          >
            <p
              class="no-item-text"
            >
              No todos
            </p>
          </div>
          <div
            class="todo-panel"
          >
            <span
              class="todo-active-count"
            >
              0
               items left
            </span>
            <div>
              <button
                class="no-style-btn selected"
                type="button"
              >
                ALL
              </button>
              <button
                class="no-style-btn "
                type="button"
              >
                COMPLETED
              </button>
              <button
                class="no-style-btn "
                type="button"
              >
                ACTIVE
              </button>
            </div>
            <button
              class="no-style-btn clear-btn"
              type="button"
            >
              Clear completed
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
